File: src/main/java/ast/ASTArrayType.java
package ast;

public class ASTArrayType extends ASTType {
    int length;

    public ASTArrayType(ASTFileLocation loc, String lit_type, int length){
        super(loc, lit_type);
        this.length = length;
    }

    public void print(int i){
        printIndent(i);
        System.out.print("ASTArrayType: ");
        System.out.print(lit_type);
        System.out.print("[");
        if(length != -1){
            System.out.print(length);
        }
        System.out.println("]");
    }
}

File: src/main/kotlin/ast/ASTArrayType.kt
package ast

class ASTArrayType(loc: ASTFileLocation, lit_type: String, var length: Int) : ASTType(loc, lit_type) {

    override fun print(i: Int) {
        printIndent(i)
        print("ASTArrayType: ")
        print(lit_type)
        print("[")
        if (length != -1) {
            print(length)
        }
        println("]")
    }
}